packages/logging/src/setup-logger-and-telemetry.spec.ts

```diff
@@ -822,4 +822,105 @@ describe('setupLoggerAndTelemetry', function () {
     expect(logOutput).to.have.lengthOf(0);
     expect(analyticsOutput).to.be.empty;
   });
+
+  it('tracks custom logging events', function () {
+    setupLoggerAndTelemetry(
+      bus,
+      logger,
+      analytics,
+      {
+        platform: process.platform,
+        arch: process.arch,
+      },
+      '1.0.0'
+    );
+    expect(logOutput).to.have.lengthOf(0);
+    expect(analyticsOutput).to.be.empty;
+
+    bus.emit('mongosh:connect', {
+      uri: 'mongodb://localhost/',
+      is_localhost: true,
+      is_atlas: false,
+      resolved_hostname: 'localhost',
+      node_version: 'v12.19.0',
+    });
```

Comment:
I'm sorry if this is silly, but as someone who's not super proficient with the mongosh testing harness, it's not clear to me why we need this as part of this test. Is it used to test that non-custom logs are still emitted or for some other reason?

Reply:
Yes, that is exactly the reason to check that all of those records are located in the same log file and the custom log is not being saved separately to keep the ordered log entries in one place to provide a clear picture for debugging.